core/src/main/java/nu/yona/server/subscriptions/service/NewDeviceRequestService.java

```diff
@@ -40,16 +40,17 @@
 	@Transactional
 	public NewDeviceRequestDto setNewDeviceRequestForUser(UUID userId, String yonaPassword, String newDeviceRequestPassword)
 	{
-		User userEntity = userService.getValidatedUserById(userId);
+		userService.updateUser(userId, ue -> {
+			User userEntity = userService.getValidatedUserById(ue.getId());
```

Comment:
Isn't this a bit strange to get the `userEntity` instead of using `ue`?

Reply:
Done